opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/OperationNameProvider.java

```diff
@@ -0,0 +1,107 @@
+package io.opentracing.contrib.jaxrs2.server;
+
+import io.opentracing.contrib.jaxrs2.internal.URIUtils;
+import java.lang.reflect.Method;
+import java.util.List;
+import java.util.Map;
+import javax.ws.rs.container.ContainerRequestContext;
+import javax.ws.rs.core.MultivaluedMap;
+
+/**
+ * @author Pavol Loffay
+ */
+public interface OperationNameProvider {
+  interface Builder {
+    OperationNameProvider build(Class<?> clazz, Method method);
+  }
+
+  String operationName(ContainerRequestContext requestContext);
+
+  /**
+   * Returns HTTP method as operation name
+   */
+  class MethodOperationName implements OperationNameProvider {
```

Comment:
Should this be called HTTPMethodOperationName, to avoid confusion with class method version?